src/Plugin/GraphQL/DataProducer/Field/EntityReferenceTrait.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\graphql\Plugin\GraphQL\DataProducer\Field;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\TranslatableInterface;
+use Drupal\graphql\GraphQL\Execution\FieldContext;
+
+/**
+ * Entity reference helpers.
+ */
+trait EntityReferenceTrait {
+
+  /**
+   * Retrieves referenced entities from the given resolver.
+   *
+   * May optionally respect bundles/language and perform access checks.
+   *
+   * @param string $type
+   *   Entity type ID.
+   * @param string|null $language
+   *   Optional. Language to be respected for retrieved entities.
+   * @param array|null $bundles
+   *   Optional. List of bundles to be respected for retrieved entities.
+   * @param bool $access
+   *   Whether to filter out inaccessible entities.
+   * @param \Drupal\Core\Session\AccountInterface|null $accessUser
+   *   User entity to check access for. Default is null.
+   * @param string $accessOperation
+   *   Operation to check access for. Default is view.
+   * @param \Closure $resolver
+   *   The resolver to execute.
+   * @param \Drupal\graphql\GraphQL\Execution\FieldContext $context
+   *   The caching context related to the current field.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface[]|null
+   *   The list of references entities. Or NULL.
+   */
+  protected function getReferencedEntities($type, $language, $bundles, $access, $accessUser, $accessOperation, \Closure $resolver, FieldContext $context) {
```

Comment:
can you add type hints for all the parameters and the return value of this method?